src/base/Notifier.sol

```diff
@@ -67,11 +67,20 @@ abstract contract Notifier is INotifier {
     {
         _positionConfigs(tokenId).setUnsubscribe();
         ISubscriber _subscriber = subscriber[tokenId];
+        if (_subscriber == NO_SUBSCRIBER) NotSubscribed.selector.revertWith();
 
         delete subscriber[tokenId];
 
-        uint256 subscriberGasLimit = block.gaslimit.calculatePortion(BLOCK_LIMIT_BPS);
-        try _subscriber.notifyUnsubscribe{gas: subscriberGasLimit}(tokenId, config, data) {} catch {}
+        if (address(_subscriber).code.length > 0) {
+            uint256 subscriberGasLimit = block.gaslimit.calculatePortion(BLOCK_LIMIT_BPS);
+
+            // require that the remaining gas is sufficient to notify the subscriber
+            // otherwise, users can select a gas limit where .notifyUnsubscribe hits OutOfGas yet the transaction/unsubscription
+            // can still succeed
+            // to account for EIP-150, condition could be 64 * gasleft() / 63 <= subscriberGasLimit
+            if ((64 * gasleft() / 63) < subscriberGasLimit) GasLimitTooLow.selector.revertWith();
+            try _subscriber.notifyUnsubscribe{gas: subscriberGasLimit}(tokenId, config, data) {} catch {}
```

Comment:
need extra eyes on this (and the natspec)

feels weirdly unintuitive that users may need to specify a very high gas limit on `.unsubscribe()` calls

@snreynolds @hensha256

Reply:
yeah it means that all users will have to send in `gasLimit > subscriberGasLimit`, even if their particular subsciber only needs like 10k in gas. Which sucks really... but I can't see any other way around it?
I think it also means that it will be really hard to run estimateGas because of this require statement.